core-primitives/ocall-api/src/lib.rs

```diff
@@ -64,11 +63,9 @@ pub trait EnclaveRpcOCallApi: Clone + Debug + Send + Sync + Default {
 
 /// trait for o-calls related to on-chain interactions
 pub trait EnclaveOnChainOCallApi: Clone + Debug + Send + Sync {
-	fn send_block_and_confirmation<SB: SignedBlock>(
-		&self,
-		confirmations: Vec<OpaqueExtrinsic>,
-		signed_blocks: Vec<SB>,
-	) -> SgxResult<()>;
+	fn send_sidechain_blocks<SB: Encode>(&self, signed_blocks: Vec<SB>) -> SgxResult<()>;
+
+	fn send_confirmations(&self, confirmations: Vec<OpaqueExtrinsic>) -> SgxResult<()>;
```

Comment:
Separated into 2 methods, and changed the `SB` constraint to `Encode`, which allowed me to get rid of the `its_primitives` dependency in this crate.

Reply:
what would you suggest then? The thing is, that the ocall API is very low level, a lot of components depend on it. And if the ocall API depends on sidechain, then everything depends on the sidechain. Plus, this is the ocall bridge/tunnel, it will get encoded anyway before it is sent out of the enclave. It's just done one step earlier than before, and now at the same level as the extrinsic as well (`OpaqueExtrinsic`). 

What we could do, to be consistent with the `send_confirmations` call, is to introduce a rust newtype `OpaqueSignedBlocks` which is a `struct(Vec<u8>)`, so we still have strong typing, but it's an encoded signed block inside. This type definition we could have in the ocall api, so we don't need the sidechain dependency. What do you think?

Reply:
Hmm, I honestly don't really know after rethinking this a few times. Strictly speaking, we might want to have an ocall-api crate in the sidechain domain. However, I wonder if this is an overkill.

I don't think we need to have an `OpaqueSidechainBlock` here. `OpaqueExtrinsics` is used here, but it actually serves a different purpose. We can't have

```
send_confirmations<Extrinsic: Encode>(&self, confirmations: Vec<Extrinsic>)
```
Because the extrinsics are themselves generic over the call, and we have extrinsics with different calls in one vec, which rust would not allow.

Maybe this is just a compromise we might need to take now. I agree with your compromise. 👍 Thanks for questioning my feedback. 😄

Reply:
It is certainly a compromise 👍 And I also think it makes sense to split the ocall/ecall bridges to support and modularize optional features like the sidechain. That will have to be done in a future refactoring ☺️